community/kernel/src/main/java/org/neo4j/kernel/AvailabilityGuard.java

```diff
@@ -281,6 +281,7 @@ private Availability availability( long millis )
             return availability;
         }
 
+        log.warn( "Database is unavailable awaiting for it to be available again for " + millis + "ms" );
```

Comment:
Seems useful. I just wonder if `warn` is the appropriate level... perhaps `info`.